Implement ticket pinning and display

wtpisaac opened the pull request:
This PR implements ticket pinning and display. Ticket pinning is performed via an action on the ticket list entry. Ticket pins records are fetched and set in the tickets list.

This approach is probably not the ideal approach; feedback on the overall approach would be appreciated.

wtpisaac committed:
Pinned tickets fetching added

wtpisaac committed:
Added post and delete methods for pinned tickets

wtpisaac committed:
Working pinned tickets on homepage

wtpisaac committed:
First pass at working team ticket pinning

wtpisaac committed:
Some minor cleanup

wtpisaac committed:
Even more minor cleanup

tdimhcsleumas commented on src/api/pinned.ts:
```diff
@@ -0,0 +1,78 @@
+  memberPk: string,
+  ticketId: number
+): Promise<PinnedTicketRecord> {
+  const pinnedTickets = await getPinnedTicketsForMember(
```
This function seems unnecessary. The api can be set up to filter based on query params and I think it would cut down on api calls. This functionality might not yet be available, so it's understandable that you wrote something like this. Keep it for now, but in the future we should move the filtering functionality to the backend.

tdimhcsleumas commented on src/components/homepage/PinnedTicket.vue:
```diff
@@ -0,0 +1,105 @@
+  emits: ["pinnedDidUpdate"],
+  setup(props, context) {
+    const number = computed(() => props.pinRecord.ticket.ticket_number);
+    const tags = computed(() => {
```
I think a v-for would be a lot cleaner here.

wtpisaac replied:
This is used within a v-for on the homepage, not sure what you mean by this.

tdimhcsleumas replied:
I'm saying that I would have used a v-for for the tags instead of building a string fro them.

tdimhcsleumas replied:
I don't understand why you're making computed properties when you could access the props directly from the template. Is this a best practice that I don't understand?

wtpisaac replied:
Ah, that's a good point, I'll implement that before merging.

wtpisaac replied:
And no, that was me forgetting props were directly accessible :P will fix

tdimhcsleumas commented on src/views/Tickets.vue:
```diff
@@ -95,6 +114,20 @@ export default defineComponent({
       );
     };
 
+    const pinRecordForTicket = (
```
This function appears to do the same thing as getPinnedTicketForTicket. I think I see why this is necessary, but the backend could probably be cleaned up to avoid needing to filter through the filters. Good enough for now.

wtpisaac replied:
I think I wrote getPinnedTicketForTicket ahead of time without thinking. It's not really useful because it has to fetch the full list of pinned tickets every time. We might use it for a ticket details page down the line, but in order to reduce unnecessary calls to the backend, I wanted to have this just filter an already refreshed pinned tickets list.

tdimhcsleumas replied:
Gotcha.

wtpisaac committed:
Added tags to pinned tickets on homepage

wtpisaac committed:
Added jumping to pinned ticket from homepage plus cleanup

Added jumping to pinned tickets from the homepage.

Also fixed some problems such as using v-for for the tags, instead
of manually constructing a tag string.

wtpisaac committed:
Removed unnecessary usage of computed properties

wtpisaac commented:
Branch should be ready for merging. Fixed a few issues that were caught in review, and reimplemented quick jumping to tickets feature.
